src/toxcore/dht/dht_friend.rs

```diff
@@ -57,54 +58,41 @@ impl DhtFriend {
     }
 
     /// get Socket Address list of a friend, a friend can have multi IP address bacause of NAT
-    pub fn get_addrs_of_clients(&self, is_ipv6_mode: bool) -> Vec<SocketAddr> {
-        self.close_nodes.nodes.iter()
-            .map(|node| node.get_socket_addr(is_ipv6_mode))
-            .filter_map(|addr| addr)
-            .collect::<Vec<SocketAddr>>()
+    pub fn get_addrs_of_clients(&self) -> Vec<SocketAddr> {
+        let mut addrs = Vec::new();
+
+        let good_nodes = self.close_nodes.nodes.iter()
+            .filter(|node| !node.assoc4.is_bad() || !node.assoc6.is_bad())
```

Comment:
I guess it should be `!node.assoc4.is_bad() && !node.assoc6.is_bad()`

Reply:
It seems correct - node is good when we received packet from at least one address, not necessary both. But anyway `DhtNode` has method `is_bad` which can be used here.